test/integration/master/synthetic_master_test.go

```diff
@@ -225,6 +228,133 @@ func TestStatus(t *testing.T) {
+				Type: appsv1beta1.RollingUpdateDeploymentStrategyType,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"foo": "bar"},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "foo",
+							Image: "foo",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	switch field {
+	case "labels":
+		labelsMap := map[string]string{}
+		for i := 0; i < size; i++ {
+			key := val + strconv.Itoa(i)
+			labelsMap[key] = val
+		}
+		deploymentObject.ObjectMeta.Labels = labelsMap
+	case "annotations":
+		annotationsMap := map[string]string{}
+		for i := 0; i < size; i++ {
+			key := val + strconv.Itoa(i)
+			annotationsMap[key] = val
+		}
+		deploymentObject.ObjectMeta.Annotations = annotationsMap
+	case "finalizers":
+		finalizerString := []string{}
+		for i := 0; i < size; i++ {
+			finalizerString = append(finalizerString, val)
+		}
+		deploymentObject.ObjectMeta.Finalizers = finalizerString
+	}
+
+	deploymentBody, err := json.Marshal(deploymentObject)
+	if err != nil {
+		t.Fatalf("Failed to marshal into json: %v", err)
+	}
+	return string(deploymentBody)
+}
+
+func TestObjectSizeResponses(t *testing.T) {
+	_, s, closeFn := framework.RunAMaster(nil)
+	defer closeFn()
+	transport := http.DefaultTransport
+
+	const DeploymentMegabyteSize = 100000
+	const DeploymentTwoMegabyteSize = 1000000
+	const RequestURL = "/apis/apps/v1beta1/namespaces/" + metav1.NamespaceDefault + "/deployments"
+
+	expectedMsgFor1MB := `etcdserver: request is too large`
+	expectedMsgFor2MB := `rpc error: code = ResourceExhausted desc = grpc: trying to send message larger than max`
+	expectedMsgForLargeAnnotation := `metadata.annotations: Too long: must have at most 262144 characters`
+
+	deployment1 := constructBody("a", DeploymentMegabyteSize, "labels", t)    // >1 MB file
+	deployment2 := constructBody("a", DeploymentTwoMegabyteSize, "labels", t) // >2 MB file
+
+	deployment3 := constructBody("a", DeploymentMegabyteSize, "annotations", t)
+
+	deployment4 := constructBody("sample/sample", DeploymentMegabyteSize, "finalizers", t)    // >1 MB file
+	deployment5 := constructBody("sample/sample", DeploymentTwoMegabyteSize, "finalizers", t) // >2 MB file
+
+	requests := []struct {
+		size            string
+		verb            string
+		URL             string
+		body            string
+		expectedMessage string
+	}{
+		{"1 MB", "POST", RequestURL, deployment1, expectedMsgFor1MB},
+		{"2 MB", "POST", RequestURL, deployment2, expectedMsgFor2MB},
+		{"1 MB", "POST", RequestURL, deployment3, expectedMsgForLargeAnnotation},
+		{"1 MB", "POST", RequestURL, deployment4, expectedMsgFor1MB},
+		{"2 MB", "POST", RequestURL, deployment5, expectedMsgFor2MB},
+	}
+
+	for _, r := range requests {
+		t.Run(fmt.Sprintf("%s %s %s %s", r.size, r.verb, r.URL, r.body), func(t *testing.T) {
+			bodyBytes := bytes.NewReader([]byte(r.body))
+			req, err := http.NewRequest(r.verb, s.URL+r.URL, bodyBytes)
```

Comment:
here we could even use the typed client in k8s.io/client-go/kubernetes for apps/v1.Deployment

Reply:
@sttts  Sure thing, will update it.

Reply:
Any update on this?

Reply:
@sttts Updated, PTAL.

Reply:
Further from future should I not squash my commits so that just my recent changes are visible? And squash them just at before submitting the PR?

Reply:
Attached [analysis](https://gist.github.com/anubhakushwaha/2095176d8b9733be1658c42fdef0ea68)

Reply:
Attached gist [here](https://gist.github.com/anubhakushwaha/2095176d8b9733be1658c42fdef0ea68)